tests/src/test/scala/common/WskTestHelpers.scala

```diff
@@ -31,6 +31,91 @@ import spray.json._
+ */
+case class ActivationResponse(result: Option[JsObject], status: String, success: Boolean)
+
+object ActivationResponse extends DefaultJsonProtocol {
+    implicit val serdes = jsonFormat3(ActivationResponse.apply)
+}
+
+/**
+ * Activation record as it is returned from the OpenWhisk service.
+ *
+ * @param activationId a String to save the ID of the activation
+ * @param logs a list of String to save the logs of the activation
+ * @param response an Object of ActivationResponse to save the response of the activation
+ * @param start an Instant to save the start time of activation
+ * @param end an Instant to save the end time of activation
+ * @param duration a Long to save the duration of the activation
+ * @param cases String to save the cause of failure if the activation fails
+ * @param annotations a list of JSON objects to save the annotations of the activation
+ */
+ case class ActivationResult (
+        activationId: String,
+        logs: Option[List[String]],
+        response: ActivationResponse,
+        start: Instant,
+        end: Instant,
+        duration: Long,
+        cause: Option[String],
+        annotations: Option[List[JsObject]]) {
+
+        def getAnnotationValue(key: String): Option[JsValue] = {
+            Try {
+                val annotation = annotations.get.filter(x => x.getFields("key")(0) == JsString(key))
+                assert(annotation.size == 1) // only one annotation with this value
+                val value = annotation(0).getFields("value")
+                assert(value.size == 1)
+                value(0)
+            } toOption
+        }
+}
+
+object ActivationResult extends DefaultJsonProtocol {
+        private implicit val instantSerdes = new RootJsonFormat[Instant] {
+            def write(t: Instant) = t.toEpochMilli.toJson
+
+            def read(value: JsValue) = Try {
+                value match {
+                    case JsNumber(i) => Instant.ofEpochMilli(i.bigDecimal.longValue)
+                    case _           => deserializationError("timetsamp malformed")
+                }
+            } getOrElse deserializationError("timetsamp malformed 2")
+        }
+
+        implicit val serdes = new RootJsonFormat[ActivationResult] {
+            private val format = jsonFormat8(ActivationResult.apply)
+
+            def write(result: ActivationResult) = format.write(result)
+
+            def read(value: JsValue) = {
+                val obj = value.asJsObject
+                obj.getFields("activationId") match {
+                    case Seq(JsString(activationId)) =>
+                        Try {
+                            val logs = obj.fields.get("logs").map(_.convertTo[List[String]])
+                            val response = obj.fields.get("response").map(_.convertTo[ActivationResponse]).get
+                            val start = obj.fields.get("start").map(_.convertTo[Instant]).get
```

Comment:
those `get`s are unsafe. Can you move `response` and `start` to the sequence above?